src/solver/conv_asm_dir_BwdWrW3x3.cpp

```diff
@@ -351,6 +351,11 @@ bool ConvAsmBwdWrW3x3::IsApplicable(const ConvolutionContext& params) const
         return false;
     if(!params.rmv.IsV2orV3())
         return false;
+
+    const auto target = params.GetStream().GetTargetProperties();
+    if(target.Xnack() && *target.Xnack())
+        return false;
+
     const std::string name = params.GetStream().GetDeviceName();
     if(!(StartsWith(name, "gfx8") || StartsWith(name, "gfx9")) || name == "gfx90a")
```

Comment:
```suggestion
    if(!(StartsWith(name, "gfx8") || StartsWith(name, "gfx9")))
```

Reply:
Okay. But we need to create tickets for individual Solvers then (or one umbrella ticket with a checklist of Solvers). May I ask you to do this while working on this PR?

Let me explain an approach in general.

Each case when an asm shader is enabled for gfx9 but specifically disabled for gfx90a should be considered as a workaround. Each workaround requires resolving. In this specific case, resolving could be one of:
- Checking & fixing the kernel (checking VGPR tuple alignment etc. as you explained above) and enabling it.
- Or, finally disabling the Solver for gfx90a with clear explanation of the reasons (in C++ comments), like
  - "We have a xDLOPs counterpart Solver for gfx90a"
  - "This Solver shows poor performance for gfx90a and we have another, faster Solvers for it, so it does not worth fixing it for gfx90a"
  - "This Solver uses some gfx9 instructions (...) that not exist on gfx90a and cannot be fixed for it" etc.

/cc @Kirpich30000